# ã‚¤ãƒ³ãƒ—ãƒƒãƒˆï¼ˆInputï¼›å…¥åŠ›ï¼‰
extends Node


# ãƒ¼ãƒ¼ãƒ¼ãƒ¼ãƒ¼ãƒ¼ãƒ¼ãƒ¼
# ãƒãƒ¼ãƒ‰ãƒ»ãƒ‘ã‚¹é–¢é€£
# ãƒ¼ãƒ¼ãƒ¼ãƒ¼ãƒ¼ãƒ¼ãƒ¼ãƒ¼

# çŒ¿å–å¾—
func monkey():
	return $"../ğŸµMonkey"


# ãƒ¼ãƒ¼ãƒ¼ãƒ¼ãƒ¼ãƒ¼ãƒ¼ãƒ¼
# å…¥åŠ›
# ãƒ¼ãƒ¼ãƒ¼ãƒ¼ãƒ¼ãƒ¼ãƒ¼ãƒ¼

# ãƒ†ã‚­ã‚¹ãƒˆãƒœãƒƒã‚¯ã‚¹ãªã©ã«ãƒ•ã‚©ãƒ¼ã‚«ã‚¹ãŒç„¡ã„ã¨ãã®ã‚­ãƒ¼å…¥åŠ›ã‚’æ‹¾ã†
#
# å­è¦ç´ ã‹ã‚‰è¦ªè¦ç´ ã®é †ã§å‘¼ã³å‡ºã•ã‚Œã‚‹ã‚ˆã†ã ã€‚
# ã“ã®ãƒ—ãƒ­ã‚°ãƒ©ãƒ ã§ã¯ã€€ãƒ«ãƒ¼ãƒˆã€€ã ã‘ã§ã€€ã‚­ãƒ¼å…¥åŠ›ã‚’æ‹¾ã†ã“ã¨ã«ã™ã‚‹
func _unhandled_key_input(event):
	# ã‚­ãƒ¼ãƒ»ã‚³ãƒ³ãƒ•ã‚£ã‚°ã®ãŸã‚ã«ã€ä½•ã‚‚ã™ã‚‹ãª
	if self.monkey().owner_node().current_state == &"WaitForKeyConfig":
		pass

	# ã‚­ãƒ¼ãƒ»ã‚³ãƒ³ãƒ•ã‚£ã‚°ä¸­ãªã®ã§ã€ä½•ã‚‚ã™ã‚‹ãª
	elif self.monkey().owner_node().current_state == &"KeyConfig":
		pass

	# ä¸»è¦ãªçŠ¶æ…‹
	elif self.monkey().owner_node().current_state == &"Main":
		self.monkey().scenario_player().input_node().on_unhandled_key_input(event)

# ãƒ†ã‚­ã‚¹ãƒˆãƒœãƒƒã‚¯ã‚¹ãªã©ã«ãƒ•ã‚©ãƒ¼ã‚«ã‚¹ãŒç„¡ã„ã¨ãã®å…¥åŠ›ã‚’ã¨ã«ã‹ãæ‹¾ã†
func _unhandled_input(event):
	# ã‚­ãƒ¼ãƒ»ã‚³ãƒ³ãƒ•ã‚£ã‚°ã®ãŸã‚ã«ä½•ã‚‚ã™ã‚‹ãªã€ã¨ã„ã†çŠ¶æ…‹
	if self.monkey().owner_node().current_state == &"WaitForKeyConfig":
		pass

	# ã‚­ãƒ¼ãƒ»ã‚³ãƒ³ãƒ•ã‚£ã‚°ã«å…¥åŠ›ã®åˆ¶å¾¡ã‚’è­²ã‚Œã€ã¨ã„ã†çŠ¶æ…‹
	elif self.monkey().owner_node().current_state == &"KeyConfig":
		self.monkey().key_config_node().on_unhandled_input(event)

	# ä¸»ãªçŠ¶æ…‹
	elif self.monkey().owner_node().current_state == &"Main":
		self.monkey().scenario_player().input_node().on_unhandled_input(event)
